packages/@ourworldindata/grapher/src/facetChart/FacetChart.tsx

```diff
@@ -342,6 +346,29 @@ export class FacetChart
         })
     }
 
+    @computed private get isSharedYAxis(): boolean {
+        // When the Y axis is uniform for all facets:
+        // - for most charts, we want to only show the axis on the left-most facet charts, and omit
+        //   it on the others
+        // - for bar charts the Y axis is plotted horizontally, so we don't want to omit it
+        return (
+            this.uniformYAxis &&
+            ![
+                ChartTypeName.StackedDiscreteBar,
+                ChartTypeName.DiscreteBar,
+            ].includes(this.chartTypeName)
+        )
+    }
+
+    @computed private get isSharedXAxis(): boolean {
+        return (
+            this.uniformXAxis &&
+            // TODO: do this for stacked area charts and line charts as well?
+            this.chartTypeName === ChartTypeName.StackedBar &&
+            this.facetCount >= 12
```

Comment:
I guess for some magic constants it's nice to extract them into a constant - can make this into `SHARED_X_AXIS_MIN_FACET_COUNT` maybe.